src/coreComponents/python/modules/pygeosx_tools_package/pygeosx_tools/well_log.py

```diff
@@ -0,0 +1,85 @@
+
+import numpy as np
+
+
+def parse_las(fname, variable_start='~CURVE INFORMATION', body_start='~A'):
+    """
+    Parse an las format log file
+
+    Args:
+        fname (str): Path to the log file
+        variable_start (str): A string that indicates the start of variable header information (default = '~CURVE INFORMATION')
+        body_start (str): a string that indicates the start of the log body (default = '~A')
+
+    Returns:
+        np.ndarray: a dict containing the values and unit definitions for each variable in the log
+    """
+    variable_names = []
+    variable_units = []
+    variable_values = []
+
+    with open(fname) as f:
+        file_location = 0
+        for line in f:
+            # Preamble
+            if (file_location == 0):
+                if variable_start in line:
+                    file_location += 1
+
+            # Variable definitions
+            elif (file_location == 1):
+                if ('#' not in line):
```

Comment:
Do you really want `'#' not in line` or do you want to check that the line is not a comment?
Splitting on `#` would be more robust then.

Reply:
It's probably more appropriate to check whether the first character is a comment.  I also need to improve the parsing of the curve information block since my original method is quite fragile.  Looking at some examples I have handy, these are the patterns that I can attempt to match:

```
~Curve Information Block
#MNEM.UNIT                 API CODE     Curve Description
#---------               -----------    ---------------------------
 DEPT.F                 00 000 000 000:  Depth
 RT90.ohmm              00 000 000 000:  90in Resistivity 2ft Res
```

```
~Curve Information Block
#MNEM.UNIT                 API CODE     Curve Description
#---------               -----------    ---------------------------
 DEPT.ft                              :DEPTH 
 GRTL1.API                             :TOTAL GAMMA RAY PASS 1
```